### PR TITLE
Use numeric_cast<mp_integer> instead of deprecated to_integer(exprt)

### PR DESCRIPTION
**Pull request by tautschnig**
This reduces the number of warnings flagged, in particular in Visual-Studio
builds.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b4a93cbc0fd95585082552d79b526497d67d45d4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89879668

File: src/util/expr_initializer.cpp [hunk 1]
```diff
@@ -172,21 +172,14 @@ exprt expr_initializert<nondet>::expr_initializer_rec(
...
-
-    if(to_integer(vector_type.size(), vector_size))
-    {
-      if(nondet)
-        return side_effect_expr_nondett(type, source_location);
-      else
-        return nil_exprt();
-    }
+    const mp_integer vector_size =
+      numeric_cast_v<mp_integer>(vector_type.size());
```
Comment (kroening) [inline src/util/expr_initializer.cpp:177]: This is a change in behavior -- intended?

File: src/util/pointer_offset_size.cpp [hunk 2]
```diff
@@ -124,15 +120,10 @@ pointer_offset_bits(const typet &type, const namespacet &ns)
       return {};
 
     // get size
-    const exprt &size=to_vector_type(type).size();
+    const mp_integer size =
+      numeric_cast_v<mp_integer>(to_vector_type(type).size());
```
Comment (kroening) [inline src/util/pointer_offset_size.cpp:125]: Same here?

**Review by kroening — APPROVED** [review]

File: src/util/expr_initializer.cpp (hunk 1 above)
Comment (tautschnig) [reply src/util/expr_initializer.cpp:177]: Yes, my apologies for not noting that in the commit message (which I have now done). In several places we had already treated vectors having constant size as an invariant. This should add consistency.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 638d91c5b94081bfb0293a5b6f7a6fd127cac56c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90146796
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by tautschnig** [comment]
Could someone with sufficient access please take a look as this may have broken TG?

**Comment by smowton** [comment]
Looked like a post-build network-related failure, but restarted the build to make sure

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 638d91c5b94081bfb0293a5b6f7a6fd127cac56c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90146796
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by smowton** [comment]
Failure is real but also present on current test-gen/develop, so nothing to worry about here

**Pull request merged**